test/tests/serial/mail_smtponly.bats

```diff
@@ -1,64 +1,50 @@
-load "${REPOSITORY_ROOT}/test/test_helper/common"
+load "${REPOSITORY_ROOT}/test/helper/common"
+load "${REPOSITORY_ROOT}/test/helper/setup"
 
-function setup_file() {
-  docker run --rm -d --name mail_smtponly \
-    -v "$(pwd)/test/test-files":/tmp/docker-mailserver-test:ro \
-    -e SMTP_ONLY=1 \
-    -e PERMIT_DOCKER=network \
-    -e OVERRIDE_HOSTNAME=mail.my-domain.com \
-    -t "${NAME}"
+BATS_TEST_NAME_PREFIX='[SMTP-Only] '
+CONTAINER_NAME='dms-test_smtp_only'
 
-  wait_for_finished_setup_in_container mail_smtponly
-  wait_for_smtp_port_in_container mail_smtponly
-}
+function setup_file() {
+  _init_with_defaults
 
-function teardown_file() {
-  docker rm -f mail_smtponly
-}
+  local CUSTOM_SETUP_ARGUMENTS=(
+    --env SMTP_ONLY=1
+    --env PERMIT_DOCKER=network
+  )
 
-#
-# configuration checks
-#
+  _common_container_setup 'CUSTOM_SETUP_ARGUMENTS'
 
-@test "checking configuration: hostname/domainname override" {
-  run docker exec mail_smtponly /bin/bash -c "cat /etc/mailname | grep my-domain.com"
-  assert_success
+  _wait_for_finished_setup_in_container "${CONTAINER_NAME}"
+  _wait_for_smtp_port_in_container
 }
 
-#
-# imap
-#
-
-@test "checking configuration: dovecot quota absent in postconf (disabled using SMTP_ONLY)" {
-  run docker exec mail_smtponly /bin/bash -c "postconf | grep 'check_policy_service inet:localhost:65265'"
-  assert_failure
-}
+function teardown_file() { _default_teardown ; }
 
-#
-# smtp
-#
-
-@test "checking smtp_only: mail send should work" {
-  run docker exec mail_smtponly /bin/sh -c "postconf smtp_host_lookup=no"
+@test "Dovecot quota absent in postconf" {
+  _run_in_container postconf
   assert_success
+  refute_output --partial "check_policy_service inet:localhost:65265'"
+}
 
-  _reload_postfix mail_smtponly
+# TODO: needs complete rework when proper DNS container is running for tests
+@test "sending mail should work" {
+  skip 'TODO: This test is absolutely broken and needs reworking!'
 
-  wait_for_smtp_port_in_container mail_smtponly
-  run docker exec mail_smtponly /bin/sh -c "nc 0.0.0.0 25 < /tmp/docker-mailserver-test/email-templates/smtp-only.txt"
+  # the value `no` is not even valid for `smtp_host_lookup`
+  # `smtp_host_lookup` seems to be deprecated too
+  _run_in_container postconf smtp_host_lookup=no
   assert_success
-  run docker exec mail_smtponly /bin/sh -c 'grep -cE "to=<user2\@external.tld>.*status\=sent" /var/log/mail/mail.log'
-  [[ ${status} -ge 0 ]]
-}
 
-#
-# PERMIT_DOCKER=network
-#
+  _reload_postfix
+  _wait_for_smtp_port_in_container
```

Comment:
If dropping `smtp_host_lookup` in this PR, the reload isn't necessary. The SMTP port should already be available as it's waited on in `setup_file()`.

```suggestion
```